crates/networking/p2p/discv4/server.rs

```diff
@@ -263,6 +271,44 @@ impl DiscoveryServer {
         }
     }
 
+    async fn enr_lookup(&mut self) -> Result<(), DiscoveryServerError> {
```

Comment:
I think it might be better to send an ENR request the moment we receive a PONG from a peer, instead of having an independent flow for these messages.

Reply:
I believe that part of the setup of that tests it to do the Ping/Pong exchange before the actual test happens, and now we are pushing an ENRRequest message after the Pong, but before anything else...